File: lesson4/python3/homework/homework4_30.py
#!/usr/bin/python3
#: lesson4/python3/homework/homework4_30.py

numb_j = int(input("Введите число чтоб узнать совершенно ли оно => "))
numb_s = 0
for numb_i in range(1, numb_j):
    if numb_j % numb_i == 0:
        numb_s += numb_i
if numb_s == numb_j:
    print("Совершенное число =>", numb_j)
else:
    print("Не совершенное число =>", numb_j)

File: lesson4/c_c++/homework/homework4_30.cpp
//: lesson4/c_c++/homework/homewor4_30.cpp
#include <iostream>

int main() {
    int numb_i, numb_j, numb_s;
    std::cout << "Введите число чтоб узнать является оно совершенным => ";
    std::cin >> numb_j;
    numb_s = 0;
    for (numb_i = 1; numb_i < numb_j; numb_i++) {
        if (numb_j % numb_i == 0) {
            numb_s += numb_i;
        }
    }
    if (numb_s == numb_j) {
        std::cout << "Совершенное число => " << numb_j << '\n';
    } else {
        std::cout << "Не совершенное число => " << numb_j << '\n';
    }
    return 0;
}
